src/org/traccar/api/MediaFilter.java

```diff
@@ -93,7 +72,17 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
             chain.doFilter(request, response);
         } catch (Exception e) {
-            formatError((HttpServletResponse) response, e);
+            HttpServletResponse httpResponse = (HttpServletResponse) response;
+            if (e instanceof SecurityException) {
```

Comment:
Actually what I meant is that instead of throwing exceptions and catching them you can just set status directly in most cases.

Reply:
We still need catch Security exceptions when checking user and device. Just made it in common way.

Reply:
Yes, but do we need to catch anything else other than security? It seems like a lot of overhead. And even if we do, `instanceof` is not the best option. You can just have separate catch clauses. But I think if we don't need exception we shouldn't use it.